src/convertToObject.js

```diff
@@ -7,6 +7,14 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  return sourceString
+    .split(';')
+    .map((keyVal) => keyVal.split(':').map((str) => str.trim()))
+    .reduce((accumulator, [key, value]) => {
+      accumulator[key] = value;
```

Comment:
The code directly assigns the value to the accumulator without checking if the key is non-empty. It would be good to add a check to ensure that empty keys are not added to the result object. This can be done by adding a condition that only assigns the key-value pair if the key is not an empty string.